packages/terra-menu/src/Menu.jsx

```diff
@@ -113,11 +114,31 @@ class Menu extends React.Component {
   push(item) {
     this.setState((prevState) => {
       const newStack = prevState.stack.slice();
-      newStack.push(item);
+      let updatedStack = {};
+      if (newStack.length - 1 >= 1) {
+        updatedStack = newStack[0].props.children.filter((list) => list.props.subMenuItems && !list.props.isDisabled);
+        updatedStack = this.recursiveFunction(updatedStack, item.key);
+      } else updatedStack = item;
```

Comment:
nit : missing braces 
```suggestion
      } else {
        updatedStack = item;
      }
```